webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/migration/CheckTableUsers.scala

```diff
@@ -0,0 +1,99 @@
+ *
+ * In accordance with the terms of section 7 (7. Additional Terms.) of
+ * the GNU General Public License version 3, the copyright holders add
+ * the following Additional permissions:
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+import bootstrap.liftweb._
+import com.normation.errors.IOResult
+import com.normation.rudder.db.Doobie
+import com.normation.zio._
+import doobie.implicits._
+import zio.interop.catz._
+
+/*
+ * During 7.3 cycle, we added the registration of users and their sessions in base.
+ * This is to allows better security logs on user sessions + allows to de-correlate rudder
+ * users from the `rudder-user.xml` file.
+ */
+class CheckTableUsers(
+    doobie: Doobie
+) extends BootstrapChecks {
+
+  import doobie._
+
+  override def description: String = "Check if database tables Users and UserSessions exist"
+
+  def createUserTables: IOResult[Unit] = {
+    val sql1 = sql"""CREATE TABLE IF NOT EXISTS Users (
+        id             text PRIMARY KEY NOT NULL CHECK (id <> '')
+      , creationDate   timestamp with time zone NOT NULL
+      , status         text NOT NULL
+      , managedBy   text NOT NULL CHECK (managedBy <> '')
+      , name           text
+      , email          text
+      , lastLogin      timestamp with time zone
+      , statusHistory  jsonb
+      , otherInfo      jsonb -- general additional user info
+    );"""
+
+    val sql2 = sql"""CREATE TABLE IF NOT EXISTS UserSessions (
+      userId       text NOT NULL CHECK (userId <> '')
+    , sessionId    text NOT NULL CHECK (sessionId <> '')
+    , creationDate timestamp with time zone NOT NULL
+    , authMethod   text
+    , permissions  text[]
+    , endDate      timestamp with time zone
+    , endCause     text
+    );"""
+
+    transactIOResult(s"Error with 'Users' table creation")(xa => sql1.update.run.transact(xa)).unit *>
+    transactIOResult(s"Error with 'v' table creation")(xa => sql2.update.run.transact(xa)).unit
+  }
```

Comment:
Some database compatibility issues that may arise : 
* The users is version 7.3 will have the tables created, and when upgrading to 8.1 they will need to add a column `tenants` to the `UserSessions` table : we should create a migration for that  
* I imagine there will be no user rolling back from version 8.1 to 7.3, but for developement I have an error reading user sessions from the table when the webapp starts (see the fix I suggested in `UserRepository` file)

Reply:
yes, we will need an additionnal migration to add the tenants column